### PR TITLE
[DSCP-226]  Use XDG_DATA_DIR for wallet key storage instead of run/tmp

### PR DESCRIPTION
**Pull request by goodsoft**
See https://serokell.slack.com/archives/CC92X27D3/p1535297402000100

### CODE REVIEW

File: wallet/src/Dscp/Wallet/KeyStorage.hs [hunk 1]
```diff
@@ -28,11 +28,17 @@ data StorageAccount = StorageAccount
 deriveJSON defaultOptions ''Storage
 deriveJSON defaultOptions ''StorageAccount
 
-storageFileName :: FilePath
-storageFileName = "run/tmp/wallet.json"
+getDataDir :: IO FilePath
+getDataDir = do
+    dir <- getXdgDirectory XdgData "disciplina"
+    createDirectoryIfMissing True dir
+    return dir
```
Comment (Martoon-00) [inline wallet/src/Dscp/Wallet/KeyStorage.hs]: We have similar logic in `Dscp.Config.AppDir`. I wouldn't complain, but keeping hardcoded application name in two different places doesn't look good.

Comment (goodsoft) [reply wallet/src/Dscp/Wallet/KeyStorage.hs]: There's very high probability of all this code going into ariadne and using its settings pretty soon

Comment (Martoon-00) [reply wallet/src/Dscp/Wallet/KeyStorage.hs]: (At least reuse `appName`)

Comment (Martoon-00) [reply wallet/src/Dscp/Wallet/KeyStorage.hs]: Meh, okay

File: wallet/src/Dscp/Wallet/KeyStorage.hs [hunk 2]
```diff
@@ -28,11 +29,17 @@ data StorageAccount = StorageAccount
...
+getDataDir :: IO FilePath
+getDataDir = do
+    dir <- getXdgDirectory XdgData appName
+    createDirectoryIfMissing True dir
+    return dir
 
-storageLockFileName :: FilePath
-storageLockFileName = "run/tmp/wallet.json.lock"
+getStoragePath :: IO FilePath
+getStoragePath = getDataDir >>= return . (<> "/wallet.json")
```
Comment (yegortimoshenko) [inline wallet/src/Dscp/Wallet/KeyStorage.hs]: Can we use something cross-platform for concatenation?

Comment (volhovm) [reply wallet/src/Dscp/Wallet/KeyStorage.hs]: @goodsoft https://www.stackage.org/haddock/lts-9.11/filepath-1.4.1.1/System-FilePath-Posix.html#v:-60--47--62-

Comment (goodsoft) [reply wallet/src/Dscp/Wallet/KeyStorage.hs]: :ok_hand:

Comment (goodsoft) [reply wallet/src/Dscp/Wallet/KeyStorage.hs]: @volhovm @yegortimoshenko https://github.com/DisciplinaOU/disciplina/pull/167

**Pull request merged**